### DISCUSSION TITLE
How to use middleware to protect api auth (pages dir)

### DISCUSSION THREAD

**Discussion by thanhtutzaw** [vercel/next.js#60705] [Help] [answered]
### Summary

I was checking cookies in my api files by calling checkCookies and return 401 unauthorized.
But I don't want to call it in my all api files manually if possiable . So I tried middle but it is not working in my first time . 
I write without matcher but it doesn't work so I added matcher code and still doesn't work . Is there anything wrong ?
I was using firebase and firebase admin sdk and store token in cookies . 

![image](https://github.com/vercel/next.js/assets/71011043/169e482e-aea7-4970-acc5-6775a911fc18)

**Comment by icyJoseph** [comment, marked as answer]
If you have a `src` directory, then you should have `src/middleware.ts` and `src/pages/`.

If there's no `src` directory, `middleware.ts` should be at the same level of `pages/`

**Reply by thanhtutzaw** [reply]
I moved middleware to outside of pages and it still doesn't work . I don't see any console log in terminal .
![image](https://github.com/vercel/next.js/assets/71011043/aeb64252-b787-4d3c-a1c3-6c37bc85fd5d)

**Reply by icyJoseph** [reply]
No, now it is inside `/types` you did not move it correctly. You have `/types/middleware.ts`, move it outside `types` as well.

**Reply by thanhtutzaw** [reply]
Oh I am sorry. Thanks for your help . I hope it will success now .
